File: Algo/Leetcode/283MoveZero.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        if(n==0) return;
        int p = 0;
        for(int i=0; i<n; i++){
            if(nums[i] != 0) nums[p++] = nums[i];
        }
        for(int i=p; i<n; i++) nums[i] = 0;
        return;
    }
};

File: Algo/Leetcode/283MoveZero.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        l = 0
        for i in range(n):
            if nums[i] != 0:
                swap(nums, i, l)
                l += 1
        for i in range(l, n):
            nums[i] = 0

        return nums


def swap(nums, i, j):
    tmp = nums[i]
    nums[i] = nums[j]
    nums[j] = tmp